jujuclient/file.go

```diff
@@ -285,7 +285,8 @@ func (s *store) SetCurrentController(name string) error {
 	if _, ok := controllers.Controllers[name]; !ok {
 		return errors.NotFoundf("controller %v", name)
 	}
-	if controllers.HasControllerChangedOnPreviousSwitch = controllers.CurrentController != name; controllers.HasControllerChangedOnPreviousSwitch {
+	controllers.HasControllerChangedOnPreviousSwitch = controllers.CurrentController != name
```

Comment:
**todo:** We have to fix it also in mem.go, I think i used the same ugly pattern.